interchain-security/pion-1/v5.0.6_upgrade.md

```diff
@@ -0,0 +1,269 @@
+
+## Upgrade steps
+
+There are 2 major ways to upgrade a node:
+
+- Manual upgrade
+- Upgrade using [Cosmovisor](https://pkg.go.dev/cosmossdk.io/tools/cosmovisor)
+    - Either by manually preparing the new binary
+    - Or by using the auto-download functionality (this is not yet recommended)
+
+If you prefer to use Cosmovisor to upgrade, some preparation work is needed before upgrade.
+
+### Make sure you are using the proper version of libwasm
+
+You can check the version you are currently using by running the following command:
+```
+$ neutrond q wasm libwasmvm-version
+
+2.1.3
+```
+The proper version is `2.1.3`.
+
+**If the version on your machine is different you MUST change it immediately!**
+
+#### Ways to change libwasmvm
+
+- Use a statically built Neutrond binary from an official Neutron release: [https://github.com/neutron-org/neutron/releases/tag/v5.0.6](https://github.com/neutron-org/neutron/releases/tag/v5.0.6)
+- If you built Neutron binary by yourself, `libwasmvm` should be loaded dynamically in your binary and somehow, the wrong `libwasmvm` library was present on your machine. You can change it by downloading the proper one and linking it to the Neutron binary manually:
+1. download a proper version of `libwasmvm`:
+
+```
+$ wget https://github.com/CosmWasm/wasmvm/releases/download/v2.1.3/libwasmvm.x86_64.so
+```
+
+2. tell the linker where to find it:
+```
+$ export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/lib/
+```
+
+3. check that libwasmvm version is correct:
+```
+$ neutrond q wasm libwasmvm-version
+2.1.3
+```
+
+### Method I: Manual Upgrade
+
+Make sure Neutron v5.0.6 is installed by either downloading a [compatible binary](https://github.com/neutron-org/neutron/releases/tag/v5.0.6), or building from source. Building from source requires **Golang 1.22.x**.
+
+Run Neutron v5.0.2 till upgrade height, the node will panic:
+
+```shell
+ERR UPGRADE "v5.0.6" NEEDED at height: 22119000: upgrade to v5.0.6 and applying upgrade "v5.0.6" at height:22119000
+```
+
+Stop the node, and switch the binary to Neutron v5.0.6 and re-start by `neutrond start`.
+
+It may take several minutes to a few hours until validators with a total sum voting power > 2/3 to complete their node upgrades. After that, the chain can continue to produce blocks.
+
+### Method II: Upgrade using Cosmovisor
+
+### Manually preparing the binary
+
+##### Preparation
+
+Install the latest version of Cosmovisor:
+
+```shell
+go install cosmossdk.io/tools/cosmovisor/cmd/cosmovisor@v1.5.0
+```
+
+**Verify Cosmovisor Version**
+```shell
+cosmovisor version
+cosmovisor version: v1.5.0
+```
+
+Create a cosmovisor folder:
+
+create a Cosmovisor folder inside `$NEUTRON_HOME` and move Neutron v5.0.2 into `$NEUTRON_HOME/cosmovisor/genesis/bin`
+
+```shell
+mkdir -p $NEUTRON_HOME/cosmovisor/genesis/bin
+cp $(which neutrond) $NEUTRON_HOME/cosmovisor/genesis/bin
+````
+
+build Neutron v5.0.6, and move neutrond v5.0.6 to `$NEUTRON_HOME/cosmovisor/upgrades/v5.0.6/bin`
```

Comment:
```suggestion
build Neutron v5.0.6, and move neutrond v5.0.6 to `$NEUTRON_HOME/cosmovisor/upgrades/v5.0.5/bin`
```
Because the upgrade name is 5.0.5 it's better to place the binary into 5.0.5 dir